DependencyInjectionContainer/DIContainer.cs

```diff
@@ -1,35 +1,48 @@
 ﻿using System.Reflection;
+using System.Runtime.CompilerServices;
 using System.Runtime.ExceptionServices;
 using DependencyInjectionContainer.Enums;
 using DependencyInjectionContainer.Exceptions;
 
 namespace DependencyInjectionContainer
 {
```

Comment:
You can start using file scoped namespaces with it file looks less nested visually
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-10.0/file-scoped-namespaces